Enhance `SceneRenderProfiler` with additional interaction tracking

dprokop opened the pull request:
This PR improves the `SceneRenderProfiler` by adding new interactions tracking and better profile management.

### Key Changes
####  Enhanced Profiler logic
- Improved `startProfile` method to handle overlapping profiles by continuing existing profiles with crumbs or properly stopping tail recording phases

#### Standardized interaction types
- Added interaction type constants for consistent profiling:
  - `REFRESH_INTERACTION` for manual and auto refresh actions
  - `TIME_RANGE_CHANGE_INTERACTION` for time range modifications
  - `FILTER_*_INTERACTION` for ad-hoc filter operations (add/remove/change/restore)
  - `VARIABLE_VALUE_CHANGED_INTERACTION` for variable value changes
  - `SCOPES_CHANGED_INTERACTION` for scopes modifications

#### Expanded Coverage
- Added profiling to previously untracked interactions: ad-hoc filter manipulations, variable value selections, and scopes changes
- Enhanced plugin crimbs tracking with specific plugin types



To test this, see PR in grafana repo: https://github.com/grafana/grafana/pull/108658

<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>6.28.3--canary.1195.16590049708.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/scenes-react@6.28.3--canary.1195.16590049708.0
  npm install @grafana/scenes@6.28.3--canary.1195.16590049708.0
  # or 
  yarn add @grafana/scenes-react@6.28.3--canary.1195.16590049708.0
  yarn add @grafana/scenes@6.28.3--canary.1195.16590049708.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

mdvictor reviewed (commented):
This looks great! I'll play with it some more on monday but overall looks awesome!

harisrozajac commented on packages/scenes/src/behaviors/SceneRenderProfiler.ts:
```diff
@@ -22,16 +22,28 @@ export class SceneRenderProfiler {
+        this.#profileStartTs = performance.now();
+        writeSceneLog(this.constructor.name, 'Profile started:', this.#profileInProgress, this.#profileStartTs);
+      } else {
+        // If there i a profile in progress but tail recording is not started, add a crumb to the current profile
```
```suggestion
        // If there is a profile in progress but tail recording is not started, add a crumb to the current profile
```

dprokop commented:
> One q: I think we should also add groupby measurements as well soon?

@mdvictor - good catch. I've added this in https://github.com/grafana/scenes/pull/1195/commits/b49899250a9e57612995ce6f8bd01e42dbc1c744. Forgot that group by has a custom renderer that wraps multi select. Moved the instrumentation to the model handler.
